flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStoreTest.java

```diff
@@ -101,13 +102,11 @@ public void testRecoverFailsIfDownloadFails() {
                         id -> {
                             throw new ExpectedTestException();
                         }));
-        final Exception exception =
-                assertThrows(
-                        Exception.class,
+        assertThatThrownBy(
                         () ->
                                 DefaultCompletedCheckpointStoreUtils.retrieveCompletedCheckpoints(
-                                        checkpointsInZooKeeper, zooKeeperCheckpointStoreUtil));
-        assertThat(exception, FlinkMatchers.containsCause(ExpectedTestException.class));
+                                        checkpointsInZooKeeper, zooKeeperCheckpointStoreUtil))
+                .satisfies(exp -> FlinkAssertions.anyCauseMatches(ExpectedTestException.class));
```

Comment:
```suggestion
                .satisfies(FlinkAssertions.anyCauseMatches(ExpectedTestException.class));
```
lambda is probably useless here.
I tried replace `ExpectedTestException` with `NumberFormatException` and test still passes...
After removal of lambda it works better

Reply:
Good catch 👍